src/UserInterface/Image/createComponentSelector.js

```diff
@@ -45,9 +59,21 @@ function createComponentSelector(store, imageUIGroup) {
   componentSelector.addEventListener(
     'change',
     action(event => {
+      console.log('changed component')
       event.preventDefault()
       event.stopPropagation()
-      store.imageUI.selectedComponentIndex = Number(event.target.value)
+      const selIdx = Number(event.target.getAttribute('data-componentIndex'))
```

Comment:
Could be cleaner here by using `dataset` exposed via [HTML5 Data Attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes).

Reply:
Thanks for pointing that out, it would be cleaner.  Will do.

Reply:
Thanks for pointing that out, I wasn't aware.  Will do.